packages/core/src/routes/connector.ts

```diff
@@ -195,7 +199,14 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
 
       // FIXME @Darcy [LOG-4696]: revisit databaseMetadata check when implementing AC
 
-      const connector = await updateConnector({ set: body, where: { id }, jsonbMode: 'replace' });
+      const { metadata: databaseMetadata, ...rest } = body;
+      const isMetadataEmpty =
+        !databaseMetadata || Object.entries(cleanDeep(databaseMetadata)).length === 0;
+      const connector = await updateConnector({
+        set: isMetadataEmpty ? rest : { metadata: cleanDeep(databaseMetadata), ...rest },
```

Comment:
what if the user want to clear custom metadata?

Reply:
Good call, fixed this and added UT for insurance.